service/worker/scanner/build_ids/scavenger.go

```diff
@@ -286,11 +292,11 @@ func (a *Activities) findBuildIdsToRemove(
 			}
 
 			if err := rateLimiter.Wait(ctx); err != nil {
-				return buildIdsToRemove, err
+				return nil, context.DeadlineExceeded
```

Comment:
Looks like `Wait` could also error on:

```go
if n > burst && limit != Inf {
		return fmt.Errorf("rate: Wait(n=%d) exceeds limiter's burst %d", n, burst)
	}
```

But we won't hit that based on the configured burst setting AFAICT.
I just don't want to rely on this assumption, I'd say that any error here should be non retryable and I would return a special "non retryable error" instead.

Reply:
Yeah, I just thought we'd mark it differently since it's not actually a deadline exceeded error and we may have other non retryable errors that we account for.

Reply:
It is a deadline exceeded error, though, we just happen to realize it before the deadline. Agreed that if/when there are other non-retryable errors, we should use a different error

Reply:
nit: It's any error that can be returned from `Wait` including burst exceeded, and context about to expire.